lib/datadog/core/telemetry/metric.rb

```diff
@@ -25,6 +21,10 @@ def initialize(name, tags: {}, common: true, interval: nil)
             @interval = interval
           end
 
+          def id
+            @id ||= "#{type}::#{name}::#{tags.join(',')}"
+          end
+
           def track(value); end
```

Comment:
Minor: Not that it has mattered a lot so far, but maybe we should add a `raise 'Must to be overridden by subclasses'` here?